numpy/lib/nanfunctions.py

```diff
@@ -1292,18 +1538,29 @@ def _nanquantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
     return result
 
 
-def _nanquantile_1d(arr1d, q, overwrite_input=False, interpolation='linear'):
+def _nanquantile_1d(arr1d, q, weights, overwrite_input=False,
+                    interpolation='linear'):
     """
     Private function for rank 1 arrays. Compute quantile ignoring NaNs.
     See nanpercentile for parameter usage
     """
+    # remove the weights that correspond to nans, leveraing _remove_nan_1d()
+    if weights is not None:
+        wgt = weights.copy().astype("f8")
```

Comment:
Better as `weights.astype(np.float64, copy=True)`, which is just `weights.astype(np.float64)` anyway

Reply:
Good point.